README.md

```diff
@@ -11,78 +11,114 @@ related-issues: (will contain links to implementation PRs)
+  - Move `pkgs/applications/misc/hello/default.nix` to `pkgs/unit/hell/hello/pkg-fun.nix`
+  - Move `pkgs/applications/misc/hello/test.nix` to `pkgs/unit/hell/hello/test.nix`
+- `pkgs.gnumake` matches all criteria:
+  The default Nix file [`pkgs/development/tools/build-managers/gnumake/default.nix`](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/development/tools/build-managers/gnumake/default.nix) transitively references only files in its own directory and no other files in nixpkgs reference `gnumake`'s files, so we can do the transformation by moving all the files from `pkgs/development/tools/build-managers/gnumake` to `pkgs/unit/gnum/gnumake`, the default Nix file ending up in `pkgs/unit/gnum/gnumake/pkg-fun.nix`.
+- Similarly `pkgs.gnumake42` in [`pkgs/development/tools/build-managers/gnumake/4.2/default.nix`](https://github.com/NixOS/nixpkgs/tree/nixos-22.05/pkgs/development/tools/build-managers/gnumake/4.2/default.nix) fulfils all criteria, even though its directory is nested in `pkgs.gnumake`'s directory, they don't reference each others files.
+- `pkgs.zsh` matches all criteria, its default Nix file is moved to `pkgs/unit/zsh/zsh/pkg-fun.nix`
+- `pkgs.emptyFile` doesn't fulfil [criteria 1](#user-content-criteria-1) (it's defined in `pkgs/build-support/trivial-builders.nix`), so it can't be moved into `pkgs/unit`
+- `pkgs.readline` doesn't fulfil [criteria 2](#user-content-criteria-2) (it's defined as an alias to `readline6`, which is itself defined as an alias to `readline63`)
+- `pkgs.readline63` doesn't fulfil [criteria 3](#user-content-criteria-3) (it [transitively references](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/development/libraries/readline/6.3.nix#L23) `link-against-ncurses.patch`, which is [also referenced](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/development/libraries/readline/7.0.nix#L30) by the definition for `pkgs.readline70`)
+- `pkgs.fetchFromGitHub` doesn't fulfil the [criteria 4](#user-content-criteria-4) (it evaluates to a function), so it can't be moved into `pkgs/unit`
+
+Here's how such a `pkgs/unit` directory structure would look like, note how all attribute names have the same level of nesting:
+```
+pkgs
+└── unit
+   ├── acpi
+   │  ├── acpi
+   │  ├── acpica-tools
+   │  ├── acpid
+   │  ┊
+   ┊
+   ├── auto
+   │  ├── autossh
+   │  ├── automirror
+   │  ├── autosuspend
+   │  ┊
+   ┊
+   ├── sl
+   │  └── sl
+   ┊
+   ├── slac
+   │  ├── slack
+   │  ├── slack-cli
+   │  └── slack-term
+   ┊
+   ├── zsh
+   │  ├── zsh
+   │  ├── zsh-autocomplete
+   │  ├── zsh-completions
+   ┊  ┊
+```
 
 # Interactions
 [interactions]: #interactions
 
-- `nix edit` is unaffected, since it uses a packages `meta.position` to get the file to edit
+- `nix edit` is unaffected, since it uses a packages `meta.position` to get the file to edit.
+  Though with this RFC `nix edit` could be updated to not have to rely on that anymore for the packages in the new hierarchy in nixpkgs.
```

Comment:
```suggestion
```

I don't think it can or should.

Reply:
If nixpkgs sufficiently exposes it as a stable API, I think it could. The benefit would be that `nix edit` could also work on packages where `meta.position` can't be evaluated, e.g. due to syntax errors. Additionally, `nix edit` might open the wrong file due to a builder not properly populating `meta.position`, this problem would also be avoided.

Reply:
We discussed this [in the last NAT meeting](https://discourse.nixos.org/t/2022-12-19-nixpkgs-architecture-team-meeting-22/24125#httpsgithubcomnixpkgs-architecturesimple-package-pathspull20discussion_r1051188387-4), I removed the sentence now